akka-stream/src/main/scala/akka/stream/impl/io/SslTlsCipherActor.scala

```diff
@@ -422,10 +425,13 @@ private[akka] class SslTlsCipherActor(settings: ActorMaterializerSettings, sslCo
 
   initialPhase(2, bidirectional)
 
-  protected def fail(e: Throwable): Unit = {
+  protected def fail(e: Throwable, closeTransport: Boolean=true): Unit = {
     if (tracing) log.debug("fail {} due to: {}", self, e.getMessage)
     inputBunch.cancel()
-    outputBunch.error(TransportOut, e)
+    if(closeTransport) {
```

Comment:
Closing the transport here prevents the SSLEngine from entering the `flushingOutbound` state when `completeOrFlush` is called after this function. This prevents the necessary TLS error messages from getting sent.